profiles/common/modules/features/nexteuropa_communities/nexteuropa_communities.module

```diff
@@ -1,7 +1,467 @@
+/**
+ * Provide a form for the private area setting.
+ */
+function nexteuropa_communities_private_area_form() {
+  $form['nexteuropa_communities_private_area'] = [
+    '#type' => 'checkbox',
+    '#title' => 'Enable the private area management for the community part.',
+    '#default_value' => variable_get('nexteuropa_communities_private_area', FALSE),
+  ];
+
+  return system_settings_form($form);
+}
+
+/**
+ * Add a new context negotiation to manage the NextEuropa communities.
+ *
+ * Implements hook_og_context_negotiation_info().
+ */
+function nexteuropa_communities_og_context_negotiation_info() {
+  $providers = [];
+
+  $providers['nexteuropa_communities'] = [
+    'name' => t('NextEuropa URL'),
+    'description' => t("Select group context for any url that start with the configured word, defaults at 'community'"),
+    'callback' => 'nexteuropa_communities_context_handler',
+  ];
+
+  return $providers;
+}
+
+/**
+ * Context handler, get the context from the URL.
+ */
+function nexteuropa_communities_context_handler() {
+  global $language;
+
+  $keyword = variable_get('nexteuropa_communities_url_keyword', 'community');
+
+  // Use the NextEuropa language split for the url to remove language suffix.
+  $path = nexteuropa_multilingual_language_negotiation_split_suffix(request_path(), [$language]);
+  $path_args = explode('/', $path[1]);
+
+  // Look for the community keyword in the path.
+  if ($path_args[0] === $keyword) {
+    $community_path = drupal_lookup_path('source', $path_args[0] . '/' . $path_args[1], $language->language);
+    // If no path was found, then it's not a community in particular.
+    if ($community_path !== FALSE) {
+      $community_path = explode('/', $community_path);
+
+      // Check that the node acts as a group.
+      $node = node_load($community_path[1]);
+      if (og_is_group('node', $node)) {
+        return [
+          'node' => [
+            $node->nid,
+          ],
+        ];
+      }
+    }
+  }
+  return [];
+}
+
+/**
+ * Implements hook_token_info().
+ */
+function nexteuropa_communities_token_info() {
+  $info = [];
+
+  $info['tokens']['site']['nexteuropa_communities_keyword'] = [
+    'name' => t('NextEuropa Communities keyword'),
+    'description' => t('A token showing the url keyword triggering the community context detection.'),
+  ];
+
+  return $info;
+}
+
+/**
+ * Implements hook_tokens().
+ */
+function nexteuropa_communities_tokens($type, $tokens, $data, $options) {
+  $replacements = [];
+
+  switch ($type) {
+    case 'site':
+      foreach ($tokens as $name => $token) {
+        switch ($name) {
+          case 'nexteuropa_communities_keyword':
+            $replacements[$token] = variable_get('nexteuropa_communities_url_keyword', 'community');;
+            break;
+        }
+      }
+      break;
+  }
+
+  return $replacements;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function nexteuropa_communities_permission() {
+  return array(
+    'access private area' => array(
```

Comment:
"access private area" is used for the node_access (access to content):
```
`    // The private area option is enabled.
    // DENY access only on public contents,
    // when the user has not the "access private area" permission.
```
It's a permission related to node access, not for the og configuration. This permission can be allowed to authenticated user for example, but authenticated user must not access to config page. 
administer group is coherent for access to 'admin/config/nexteuropa_communities' and 'admin/config/nexteuropa_communities/nexteuropa_private_area'